docs(core): add troubelshooting guide for convert-to-inferred migrations (#26630)
nrwl/nx: docs/shared/guides/troubleshoot-convert-to-inferred.md

# Troubleshoot Convert to Inferred Migration

> Nx comes with plugins that automatically [infer tasks](/concepts/inferred-tasks) (i.e.g Project Crystal) for your projects based on the configuration of different tools. Inference plugins comes with many benefits, such as reduced boilerplate, and access to features such as [task splitting](/ci/features/split-e2e-tasks). To make the transition easier for existing projects that are not yet using inference plugins, many plugins provide the `conver-to-inferred` generator that will switch from executor-based tasks to inferred tasks.

Reviewer: ```suggestion
Nx comes with plugins that automatically [infer tasks](/concepts/inferred-tasks) (i.e. Project Crystal) for your projects based on the configuration of different tools. Inference plugins comes with many benefits, such as reduced boilerplate, and access to features such as [task splitting](/ci/features/split-e2e-tasks). To make the transition easier for existing projects that are not yet using inference plugins, many plugins provide the `conver-to-inferred` generator that will switch from executor-based tasks to inferred tasks.
```

Should this be `i.e.` or `e.g.` ?

Reviewer: ```suggestion
Nx comes with plugins that automatically [infer tasks](/concepts/inferred-tasks) (i.e.g Project Crystal) for your projects based on the configuration of different tools. Inference plugins comes with many benefits, such as reduced boilerplate, and access to features such as [task splitting](/ci/features/split-e2e-tasks). To make the transition easier for existing projects that are not yet using inference plugins, many plugins provide the `convert-to-inferred` generator that will switch from executor-based tasks to inferred tasks.
```

> Nx comes with plugins that automatically [infer tasks](/concepts/inferred-tasks) (i.e.g Project Crystal) for your projects based on the configuration of different tools. Inference plugins comes with many benefits, such as reduced boilerplate, and access to features such as [task splitting](/ci/features/split-e2e-tasks). To make the transition easier for existing projects that are not yet using inference plugins, many plugins provide the `conver-to-inferred` generator that will switch from executor-based tasks to inferred tasks.

Reviewer 2: Combining Colum's suggestions and mine:

```suggestion
Nx comes with plugins that automatically [infer tasks](/concepts/inferred-tasks) (i.e. Project Crystal) for your projects based on the configuration of different tools. Inference plugins come with many benefits, such as reduced boilerplate and access to features such as [task splitting](/ci/features/split-e2e-tasks). To make the transition easier for existing projects that are not yet using inference plugins, many plugins provide the `convert-to-inferred` generator that will switch from executor-based tasks to inferred tasks.
```

> To see a list of the available migration generator, run:

Reviewer: ```suggestion
To see a list of the available migration generators, run:
```

```shell

> nx g conver-to-inferred

Reviewer: ```suggestion
nx g convert-to-inferred
```

```

> This will prompt to choose a plugin to run the migration for.

Reviewer 2: ```suggestion
This will prompt you to choose a plugin to run the migration for.
```

> Although the `convert-to-inferred` generator should work for most projects, there are situations that additional changes to be done by hand. If you run into issues that are not covered on this page, please open an issue on [GitHub](https://github.com/nrwl/nx/issues).

Reviewer: ```suggestion
Although the `convert-to-inferred` generator should work for most projects, there are situations that require additional changes to be done by hand. If you run into issues that are not covered on this page, please open an issue on [GitHub](https://github.com/nrwl/nx/issues).
```

## Error: The nx plugin did not find a project inside...

> This error occurs when configuration file matching the tooling cannot be found. For example, Vite works with `vite.config.ts` (or `.js`, `.cts`, `.mts`, etc.). If you've named your configuration file to something unconventional, you must rename it back to the standard naming convention before running the migration generator again.

Reviewer: ```suggestion
This error occurs when a configuration file matching the tooling cannot be found. For example, Vite works with `vite.config.ts` (or `.js`, `.cts`, `.mts`, etc.). If you've named your configuration file to something unconventional, you must rename it back to the standard naming convention before running the migration generator again.
```

For example, if you have a `apps/demo/vite.custom.ts` file and are running `nx g @nx/vite:convert-to-inferred`, you must first rename the file to `apps/demo/vite.config.ts` before running the generator.

> ## Vite: Unsupported `proxyConfig` Option

Reviewer 2: These plugin specific errors should probably be located on the `convert-to-inferred` generator page for that plugin.  You could link to them from here.

Author: I think we can do a follow-up on this.
